src/anemoi/inference/decorators.py

```diff
@@ -50,43 +51,16 @@ def __init__(self, name: str):
         """
         self.name = name
 
-    def __call__(self, func: F) -> F:
-        """Decorate the function to set the main argument.
-
-        Parameters
-        ----------
-        func : Callable[..., Any]
-            The function to decorate.
-
-        Returns
-        -------
-        Callable[..., Any]
-            The decorated function.
-        """
-
-        @wraps(func)
-        def wrapped(context: Context, main: object = MARKER, *args: Any, **kwargs: Any) -> Any:
-            """Decorator function to set the main argument.
-
-            Parameters
-            ----------
-            context : Any
-                The context in which the function is called.
-            main : Any, optional
-                The main argument value, by default MARKER.
-            *args : tuple
-                Additional positional arguments.
-            **kwargs : dict
-                Additional keyword arguments.
+    def __call__(self, cls: F) -> F:
+        """Decorate the class to set the main argument."""
 
-            Returns
-            -------
-            Any
-                The result of the decorated function.
-            """
-            if main is not MARKER:
-                kwargs[self.name] = main
+        if not isinstance(cls, type):
+            raise TypeError("main_argument can only be used to decorate classes")
```

Comment:
```suggestion
            raise TypeError("'main_argument' can only be used to decorate classes")
```
Semantics, so happy to ignore